### ISSUE TITLE
Search Functionality is not fetching any results on 404 Page

### ISSUE THREAD

**Issue by Codex-Bugmenot** [ethereum/ethereum-org-website#12518]
### Describe the bug

The Search Functionality (ctrl + K) fetches results based on query typed in the search field  but in a 404 page the results are shown as "No Results Found" for every query

### To reproduce

1. Go to any 404 page , i used "https://ethereum.org/en/1"
2. Click on Search Bar or hold (ctrl + K)
3. Search for any known search term in the website
4. Observe that is fetches no results no matter the query

### Expected behavior

The Search Functionality should Fetch Results in a 404 i.e in case some one really ends up in a 404 page they should be able to find their way back through search

### Screenshots

![Results](https://github.com/ethereum/ethereum-org-website/assets/59927099/76d4481f-fc3f-462f-b9da-6fba32f6efbe)
![NoResults](https://github.com/ethereum/ethereum-org-website/assets/59927099/b1f2f041-2f22-47c8-a47b-17e501bcae95)

### Desktop (please complete the following information)

- Browser : Brave (Chromium Build)

### Smartphone (please complete the following information)

- Browser : Brave (Chromium Build)

### Would you like to work on this issue?

- [ ] Yes
- [X] No

**Comment by tomasgrusz** [comment]
Would like to add that the language change doesn't work either, which leads me to suspect that the DEFAULT language messes up the logic discussed in this issue, as well as in issues [#12471](https://github.com/ethereum/ethereum-org-website/issues/12471) and [#12292](https://github.com/ethereum/ethereum-org-website/issues/12292).

**Comment by wackerow** [comment]
Yeah.. I agree @tomasgrusz... think we're starting to see the edge cases from that `default` locale decision. 

The search component utilizes Algolia, which does include a "facet" option that breaks up the index according to the `lang` attribute assigned to the `html` element... in the case of the 404 page, this is being set to `default` instead of `en`. We're then querying for a part of our index with `lang="default"`, which doesn't exist.

I'll assign myself for now, but also cc: @pettinarip to keep you in the loop here. I'll take a look at Algolia and see if there is anything we can do on that end to try and use the `en` index as the default.

**Comment by corwintines** [comment]
This issue seems to have been resolved, so going to close this out.
